mon-pix/app/services/session.js

```diff
@@ -92,39 +92,25 @@ export default class CurrentSessionService extends SessionService {
     const domain = this.currentDomain.getExtension();
 
     if (domain === FRANCE_TLD) {
-      await this.locale.setLocale(DEFAULT_LOCALE);
+      this.locale.setLocale(DEFAULT_LOCALE);
 
       if (!this.locale.hasLocaleCookie()) {
         this.locale.setLocaleCookie(FRENCH_FRANCE_LOCALE);
       }
       return;
     }
 
+    if (localeFromQueryParam) {
+      this.locale.setLocale(localeFromQueryParam);
+      return;
+    }
+
     if (isUserLoaded) {
-      if (localeFromQueryParam) {
-        this.currentUser.user.lang = localeFromQueryParam;
-        try {
-          await this.currentUser.user.save({ adapterOptions: { lang: this.currentUser.user.lang } });
-          await this.locale.setLocale(this.currentUser.user.lang);
-        } catch (error) {
-          const status = get(error, 'errors[0].status');
-          if (status === '400') {
-            this.currentUser.user.rollbackAttributes();
-            await this.locale.setLocale(this.currentUser.user.lang);
-          } else {
-            throw error;
-          }
-        }
-      } else {
-        await this.locale.setLocale(this.currentUser.user.lang);
-      }
-    } else {
-      if (localeFromQueryParam) {
-        await this.locale.setLocale(localeFromQueryParam);
-      } else {
-        await this.locale.setLocale(DEFAULT_LOCALE);
-      }
+      this.locale.setLocale(this.currentUser.user.lang);
+      return;
     }
+
+    this.locale.setLocale(DEFAULT_LOCALE);
```

Comment:
cf. https://github.com/1024pix/pix/blob/dev/orga/app/services/session.js et https://github.com/1024pix/pix/blob/dev/certif/app/services/session.js j'aurais mis plutôt `FRENCH_INTERNATIONAL_LOCALE` que `DEFAULT_LOCALE`.


```suggestion
      this.locale.setLocale(DEFAULT_LOCALE);
```

Reply:
Idem que ma réponse précédente.

Reply:
👌